llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2038,6 +2038,54 @@ static Instruction *foldSelectGEP(GetElementPtrInst &GEP,
   return SelectInst::Create(Cond, NewTrueC, NewFalseC, "", nullptr, Sel);
 }
 
+// Canonicalization:
+// gep T, (gep i8, base, C1), (Index +nsw C2) into
+// gep T, (gep i8, base, C1 + C2 * sizeof(T)), Index
+static Instruction *canonicalizeGEPOfConstGEPI8(GetElementPtrInst &GEP,
+                                                GEPOperator *Src,
+                                                InstCombinerImpl &IC) {
+  if (GEP.getNumIndices() != 1)
+    return nullptr;
+  auto &DL = IC.getDataLayout();
+  if (!Src->getSourceElementType()->isIntegerTy(8) ||
+      !Src->hasAllConstantIndices())
```

Comment:
Better to use accumulateConstantOffset() instead of making it i8 specific.

Reply:
Okay, you already do this below. We should just drop the i8 check here.

Reply:
I guess this doesn't matter anymore, now that the canonicalization has landed. Could even use `m_PtrAdd` matcher now to get C1.